modules/foodcenter/Classes/ProductList.php

```diff
@@ -108,7 +108,11 @@ public function add_product($id, $opt)
 
                 // If it is not the same product, get the last key
                 end($this->product);
-                $key_array = each($this->product);
+                $key_array[1] = current($this->product);
```

Comment:
This (and the other instance) looks strange.

I think these should rather be initialized at beginning of the method, shouldn't they?

Reply:
I agree that this looks strange, but the new logic reflects the same behavior like `each()`:

See the output in different PHP Versions: https://3v4l.org/p9Pof

See a comment in the PHP docs: https://www.php.net/manual/de/function.each.php#126076

Reply:
Still looks strange to me, but I trust the documentation ;)
Guess I also dated myself with this, did stop to follow language development actively at some point